app/controllers/users_controller.rb

```diff
@@ -1,43 +1,53 @@
 class UsersController < ApplicationController
+  before_action :auth_hash, only: [:create]
```

Comment:
You're calling this helper method but not storing the auth hash return value anywhere.